MicroSplit config for lightning api draft

CatEek opened the pull request:
## Description

<!-- This section provides the necessary background and information for reviewers to
understand the code and have the correct mindset when examining changes. -->

> [!NOTE]  
> **tldr**: <!-- Write a one sentence summary. -->

Lighnting API config fucntion for MS, based on minimal example for HDN. Proposed structure repeats that one of Careamist API, but keeps the model, dataloader and trainer setup separate.


### Background - why do we need this PR?

<!-- What problem are you solving? Describe in a few sentences the state before
this PR. Use code examples if useful. -->
Extends on recently proposed configuration for HDN (#511 )
**Note: Example below contains CAREAmics style datamodule, which isn't a part of this PR. 
```
config = create_microsplit_configuration(
    experiment_name="ms_test",
    data_type="tiff",
    axes="SYX",
    z_dims=[32] * 4,
    patch_size=(128, 128),
    batch_size=64,
    num_epochs=5,
    predict_logvar="pixelwise",
    train_dataloader_params={"num_workers": 4},
    val_dataloader_params={"num_workers": 4},
    logger=None
)
model = VAEModule(config.algorithm_config)
data_module = create_train_datamodule(
    train_data=train_path,
    val_data=val_path,
    data_type=config.data_config.data_type,
    patch_size=config.data_config.patch_size,
    axes=config.data_config.axes,
    batch_size=config.data_config.batch_size,
    transforms=[],
    train_dataloader_params=config.data_config.train_dataloader_params,
    val_dataloader_params=config.data_config.val_dataloader_params,
)
trainer = Trainer()
trainer.fit()
```
### Overview - what changed?

<!-- What aspects and mechanisms of the code base changed? Describe only the general 
idea and overarching features. -->

Reuses configuration function from Careamist API, but creates all modules separately.

## How has this been tested?

<!-- Describe the tests that you ran to verify your changes. This can be a short 
description of the tests added to the PR or code snippet to reproduce the change
in behaviour. -->

Performance haven't been tested

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [ ] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

CatEek committed:
microsplit config + tests

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
tests

jdeschamps commented on src/careamics/config/algorithms/microsplit_algorithm_model.py:
```diff
@@ -0,0 +1,103 @@
+MICROSPLIT = "MicroSplit"
+
+MICROSPLIT_DESCRIPTION = """MicroSplit is a self-supervised deep learning method for
+microscopy image restoration that combines the strengths of both denoising and
```
```suggestion
microscopy image splitting that combines the strengths of both denoising and
```

jdeschamps commented on src/careamics/config/algorithms/vae_algorithm_model.py:
```diff
@@ -40,14 +40,17 @@ class VAEBasedAlgorithm(BaseModel):
     # TODO: Use supported Enum classes for typing?
     #   - values can still be passed as strings and they will be cast to Enum
-    algorithm: Literal["musplit", "denoisplit"]
+    algorithm: Literal["hdn", "microsplit", "musplit", "denoisplit"]
```
```suggestion
    algorithm: Literal["hdn", "microsplit", "denoisplit"]
```

jdeschamps commented on src/careamics/config/algorithms/vae_algorithm_model.py:
```diff
@@ -64,22 +67,25 @@ def algorithm_cross_validation(self: Self) -> Self:
-                )
+                    f"Algorithm {self.algorithm} only supports loss `microsplit`."
+                )  # TODO Update losses configs
+
```
Here, I am a bit confused because the loss can be any of musplit, denoisplit and denoisplit_musplit, but the error message only says `microsplit`. Which is it?

CatEek replied:
I think we should drop variations and only leave microsplit

jdeschamps replied:
I agree!

jdeschamps commented on src/careamics/config/configuration_factories.py:
```diff
@@ -1273,3 +1284,575 @@ def create_n2v_configuration(
+
+
+def _create_vae_based_algorithm(
+    algorithm: Literal["hdn"],
```
```suggestion
    algorithm: Literal["hdn", "microsplit"],
```

jdeschamps commented on src/careamics/config/configuration_factories.py:
```diff
@@ -1273,3 +1284,575 @@ def create_n2v_configuration(
+        If required parameters are missing for the specified loss_type.
+    """
+    # gaussian likelihood
+    if loss_type in ["musplit", "denoisplit_musplit"]:
```
Not a comment to the code, but can we get a summary written somewhere (notion?) of the algorithms (musplit, denoisplit, denoisplit_musplit) and there differences in a table for instance?

It is hard to review otherwise, and we will need to write it on the website anyway!

CatEek replied:
https://www.notion.so/MicroSplit-Integration-26aa07b1e30c80428cdcde6a9f94bbd2

jdeschamps commented on src/careamics/config/configuration_factories.py:
```diff
@@ -1273,3 +1284,575 @@ def create_n2v_configuration(
+    predict_logvar: Literal["pixelwise"] | None = None,
+    logvar_lowerbound: Union[float, None] = None,
+    logger: Literal["wandb", "tensorboard", "none"] = "none",
+    model_params: dict | None = None,
```
As in https://github.com/CAREamics/careamics/pull/511, is `model_params` of any use? Aren't all model parameters set explicitely here?

CatEek replied:
removed

jdeschamps commented on src/careamics/config/configuration_factories.py:
```diff
@@ -1273,3 +1284,575 @@ def create_n2v_configuration(
+    )
+
+
+def create_microsplit_configuration(
```
Missing loss parameter?

CatEek replied:
Shouldn't be there ultimately

jdeschamps commented on src/careamics/config/configuration_factories.py:
```diff
@@ -1273,3 +1284,575 @@ def create_n2v_configuration(
+    transform_list = _list_spatial_augmentations(augmentations)
+
+    loss_config = LVAELossConfig(
+        loss_type="denoisplit_musplit", denoisplit_weight=0.9, musplit_weight=0.1
```
If the weights should sum to `1` then I would add a validation in the Pydantic model in this PR.

CatEek replied:
Don't know yet? Added todo

jdeschamps commented on src/careamics/config/loss_model.py:
```diff
@@ -35,7 +35,9 @@ class LVAELossConfig(BaseModel):
 
-    loss_type: Literal["musplit", "denoisplit", "denoisplit_musplit"]
+    loss_type: Literal[
+        "hdn", "microsplit", "musplit", "denoisplit", "denoisplit_musplit"
```
```suggestion
        "hdn", "microsplit", "denoisplit", "denoisplit_musplit"
```

CatEek replied:
Probably outdated

jdeschamps commented on src/careamics/config/support/supported_losses.py:
```diff
@@ -21,8 +21,9 @@ class SupportedLoss(str, BaseEnum):
     # PN2V = "pn2v"
-    # HDN = "hdn"
+    HDN = "hdn"
     MUSPLIT = "musplit"
+    MICROSPLIT = "microsplit"
```
```suggestion
    MICROSPLIT = "microsplit"
```

jdeschamps commented on src/careamics/config/support/supported_losses.py:
```diff
@@ -21,8 +21,9 @@ class SupportedLoss(str, BaseEnum):
     MUSPLIT = "musplit"
+    MICROSPLIT = "microsplit"
     DENOISPLIT = "denoisplit"
     DENOISPLIT_MUSPLIT = "denoisplit_musplit"
```
I would rename `denoisplit_musplit` into `denoisplit_microsplit` everywhere for consistency. There are a bunch of them laying around.

```suggestion
    DENOISPLIT_MICROSPLIT = "denoisplit_microsplit"
```

CatEek replied:
Should be removed

jdeschamps commented on src/careamics/lightning/lightning_module.py:
```diff
@@ -480,35 +504,55 @@ def predict_step(self, batch: Tensor, batch_idx: Any) -> Any:
+        mmse_list = []
+        for _ in range(self.algorithm_config.mmse_count):
+            # apply test-time augmentation if available
+            if self._trainer.datamodule.prediction_config.tta_transforms:
```
Actually, I think we should remove the test time augmentations for now, they are faulty (users can pass test-time augmentations the model was not trained on) and I am not sure that they are applied correctly.

CatEek replied:
That's not specifically a problem, but we can discuss separately

jdeschamps commented on src/careamics/losses/loss_factory.py:
```diff
@@ -59,6 +64,9 @@ def loss_factory(loss: Union[SupportedLoss, str]) -> Callable:
+        return hdn_loss
+
     elif loss == SupportedLoss.MUSPLIT:
         return musplit_loss
```
Shouldn't it be `SupportedLoss.MICROSPLIT`?

CatEek replied:
Losses need to be refactored, we should only have microsplit ultimately

jdeschamps reviewed (changes requested):
I have not looked at the Lightning module part because I don't think I can understand it, I also skipped the tests.

I have only very minimal suggestions.

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.16.0,>=0.12.3 to >=0.12.3,<=0.17.3 (#529)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h3>Docs</h3>
<ul>
<li>📝 Remove duplicated line in <code>environment-variables.md</code>.
PR <a
href="https://redirect.github.com/fastapi/typer/pull/1277">#1277</a> by
<a href="https://github.com/neirzhei"><code>@​neirzhei</code></a>.</li>
<li>📝 Fix reference to <code>count</code> parameter in the
documentation. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1201">#1201</a> by
<a
href="https://github.com/PokkaKiyo"><code>@​PokkaKiyo</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.11.13 to 0.12.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1276">#1276</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1247">#1247</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Bump actions/checkout from 4 to 5. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1271">#1271</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump mkdocs-macros-plugin from 1.3.7 to 1.3.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1272">#1272</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump tiangolo/latest-changes from 0.3.2 to 0.4.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1265">#1265</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump pillow from 11.2.1 to 11.3.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1249">#1249</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1206">#1206</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/a5e41c53783558591923337d6a9579f7f64e1152"><code>a5e41c5</code></a>
🔖 Release version 0.17.3</li>
<li><a
href="https://github.com/fastapi/typer/commit/45cae190db45817343b00ff8b403b135f34c658f"><code>45cae19</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/0b8fa09bb6b5ed5f9ed5e26c0e8a7c4eb984226d"><code>0b8fa09</code></a>
✨ Allow annotated parsing with a subclass of <code>Path</code> (<a
href="https://redirect.github.com/fastapi/typer/issues/1183">#1183</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f0334c2a1081b3ffcb99bc4c42d85cc9d360c0e0"><code>f0334c2</code></a>
🔖 Release version 0.17.2</li>
<li><a
href="https://github.com/fastapi/typer/commit/73aca3bf87dfb5331486a603a01a1980299f38a5"><code>73aca3b</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/cdc29e27eff577403fc46b4f2d3dc912d975ad66"><code>cdc29e2</code></a>
🐛 Avoid printing <code>default: None</code> in the help section when
using Rich (<a
href="https://redirect.github.com/fastapi/typer/issues/1120">#1120</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/50b1018a81cb3f99534683e71b366df90cfe5d13"><code>50b1018</code></a>
🔖 Release version 0.17.1</li>
<li><a
href="https://github.com/fastapi/typer/commit/4e811192ac56edd1921de8b846ddf6145b9fe27e"><code>4e81119</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4d1cfd2698ec8e4809e58b51b1b32398970aa512"><code>4d1cfd2</code></a>
🐛 Fix markdown formatting in <code>--help</code> output (<a
href="https://redirect.github.com/fastapi/typer/issues/815">#815</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/83eaba8d6822023b4d46212c8ef75bbd707cc49d"><code>83eaba8</code></a>
🔖 Release version 0.17.0</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.3">compare
view</a></li>
</ul>
</details>
<br />

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.16.0,>=0.12.3 to >=0.12.3,<=0.17.3 (#529)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h3>Docs</h3>
<ul>
<li>📝 Remove duplicated line in <code>environment-variables.md</code>.
PR <a
href="https://redirect.github.com/fastapi/typer/pull/1277">#1277</a> by
<a href="https://github.com/neirzhei"><code>@​neirzhei</code></a>.</li>
<li>📝 Fix reference to <code>count</code> parameter in the
documentation. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1201">#1201</a> by
<a
href="https://github.com/PokkaKiyo"><code>@​PokkaKiyo</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.11.13 to 0.12.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1276">#1276</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1247">#1247</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Bump actions/checkout from 4 to 5. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1271">#1271</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump mkdocs-macros-plugin from 1.3.7 to 1.3.9. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1272">#1272</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump tiangolo/latest-changes from 0.3.2 to 0.4.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1265">#1265</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump pillow from 11.2.1 to 11.3.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1249">#1249</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1206">#1206</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/a5e41c53783558591923337d6a9579f7f64e1152"><code>a5e41c5</code></a>
🔖 Release version 0.17.3</li>
<li><a
href="https://github.com/fastapi/typer/commit/45cae190db45817343b00ff8b403b135f34c658f"><code>45cae19</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/0b8fa09bb6b5ed5f9ed5e26c0e8a7c4eb984226d"><code>0b8fa09</code></a>
✨ Allow annotated parsing with a subclass of <code>Path</code> (<a
href="https://redirect.github.com/fastapi/typer/issues/1183">#1183</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f0334c2a1081b3ffcb99bc4c42d85cc9d360c0e0"><code>f0334c2</code></a>
🔖 Release version 0.17.2</li>
<li><a
href="https://github.com/fastapi/typer/commit/73aca3bf87dfb5331486a603a01a1980299f38a5"><code>73aca3b</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/cdc29e27eff577403fc46b4f2d3dc912d975ad66"><code>cdc29e2</code></a>
🐛 Avoid printing <code>default: None</code> in the help section when
using Rich (<a
href="https://redirect.github.com/fastapi/typer/issues/1120">#1120</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/50b1018a81cb3f99534683e71b366df90cfe5d13"><code>50b1018</code></a>
🔖 Release version 0.17.1</li>
<li><a
href="https://github.com/fastapi/typer/commit/4e811192ac56edd1921de8b846ddf6145b9fe27e"><code>4e81119</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4d1cfd2698ec8e4809e58b51b1b32398970aa512"><code>4d1cfd2</code></a>
🐛 Fix markdown formatting in <code>--help</code> output (<a
href="https://redirect.github.com/fastapi/typer/issues/815">#815</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/83eaba8d6822023b4d46212c8ef75bbd707cc49d"><code>83eaba8</code></a>
🔖 Release version 0.17.0</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.3">compare
view</a></li>
</ul>
</details>
<br />

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.17.3,>=0.12.3 to >=0.12.3,<=0.17.4 (#531)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/1b50007284ccb38e71a5752b09b2cd3ca053d98e"><code>1b50007</code></a>
🔖 Release version 0.17.4</li>
<li><a
href="https://github.com/fastapi/typer/commit/90aab51837bbf0cf11c314c0fcbdd1e55f84d29a"><code>90aab51</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/263e42769f0d54b986275a800227c717c18660ea"><code>263e427</code></a>
🐛 Make sure <code>rich.markup</code> is imported when rendering help
text (<a
href="https://redirect.github.com/fastapi/typer/issues/1290">#1290</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f99defb7cad881190cae7c60c617bf4e1619bc43"><code>f99defb</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/f116fca2485a3d59cd879a07a74b8b92c028b4a3"><code>f116fca</code></a>
⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0 (<a
href="https://redirect.github.com/fastapi/typer/issues/1292">#1292</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/9c5c5437dde460c2cf712df4198ea9eeaab64c28"><code>9c5c543</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4c2129325a1014241a663f99f482a12af6868afc"><code>4c21293</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/bdf471038d6198bb67638b17e8d03fe0a07045d8"><code>bdf4710</code></a>
⬆ Bump ruff from 0.12.10 to 0.12.11 (<a
href="https://redirect.github.com/fastapi/typer/issues/1283">#1283</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/e8e472d1048c097cc69f5152d8146c4ba4d67a00"><code>e8e472d</code></a>
⬆ [pre-commit.ci] pre-commit autoupdate (<a
href="https://redirect.github.com/fastapi/typer/issues/1288">#1288</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/075eb3020706c136c56a1899d8d02d50d022ec9a"><code>075eb30</code></a>
📝 Update release notes</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.4">compare
view</a></li>
</ul>
</details>
<br />


Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Update typer requirement from <=0.17.3,>=0.12.3 to >=0.12.3,<=0.17.4 (#531)

Updates the requirements on [typer](https://github.com/fastapi/typer) to
permit the latest version.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/releases">typer's
releases</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/fastapi/typer/blob/master/docs/release-notes.md">typer's
changelog</a>.</em></p>
<blockquote>
<h2>0.17.4</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Make sure <code>rich.markup</code> is imported when rendering help
text. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1290">#1290</a> by
<a
href="https://github.com/g-arjones"><code>@​g-arjones</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1292">#1292</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ Bump ruff from 0.12.10 to 0.12.11. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1283">#1283</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1288">#1288</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>👷 Set permissions for conflict detector workflow. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1287">#1287</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
<li>👷 Detect and label merge conflicts on PRs automatically. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1286">#1286</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<h2>0.17.3</h2>
<h3>Features</h3>
<ul>
<li>✨ Allow annotated parsing with a subclass of <code>Path</code>. PR
<a href="https://redirect.github.com/fastapi/typer/pull/1183">#1183</a>
by <a
href="https://github.com/emfdavid"><code>@​emfdavid</code></a>.</li>
</ul>
<h2>0.17.2</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing <code>default: None</code> in the help section when
using Rich. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1120">#1120</a> by
<a
href="https://github.com/mattmess1221"><code>@​mattmess1221</code></a>.</li>
</ul>
<h2>0.17.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Fix markdown formatting in <code>--help</code> output. PR <a
href="https://redirect.github.com/fastapi/typer/pull/815">#815</a> by <a
href="https://github.com/gar1t"><code>@​gar1t</code></a>.</li>
</ul>
<h2>0.17.0</h2>
<h3>Features</h3>
<ul>
<li>⚡️ Lazy-load <code>rich_utils</code> to reduce startup time. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1128">#1128</a> by
<a href="https://github.com/oefe"><code>@​oefe</code></a>.</li>
</ul>
<h3>Internal</h3>
<ul>
<li>⬆ Bump ruff from 0.12.9 to 0.12.10. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1280">#1280</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
<li>⬆ [pre-commit.ci] pre-commit autoupdate. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1281">#1281</a> by
<a
href="https://github.com/apps/pre-commit-ci"><code>@​pre-commit-ci[bot]</code></a>.</li>
<li>⬆ Update pytest-sugar requirement from &lt;1.1.0,&gt;=0.9.4 to
&gt;=0.9.4,&lt;1.2.0. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1279">#1279</a> by
<a
href="https://github.com/apps/dependabot"><code>@​dependabot[bot]</code></a>.</li>
</ul>
<h2>0.16.1</h2>
<h3>Fixes</h3>
<ul>
<li>🐛 Avoid printing additional output with
<code>no_args_is_help=True</code> and Click 8.2. PR <a
href="https://redirect.github.com/fastapi/typer/pull/1278">#1278</a> by
<a href="https://github.com/svlandeg"><code>@​svlandeg</code></a>.</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/fastapi/typer/commit/1b50007284ccb38e71a5752b09b2cd3ca053d98e"><code>1b50007</code></a>
🔖 Release version 0.17.4</li>
<li><a
href="https://github.com/fastapi/typer/commit/90aab51837bbf0cf11c314c0fcbdd1e55f84d29a"><code>90aab51</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/263e42769f0d54b986275a800227c717c18660ea"><code>263e427</code></a>
🐛 Make sure <code>rich.markup</code> is imported when rendering help
text (<a
href="https://redirect.github.com/fastapi/typer/issues/1290">#1290</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/f99defb7cad881190cae7c60c617bf4e1619bc43"><code>f99defb</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/f116fca2485a3d59cd879a07a74b8b92c028b4a3"><code>f116fca</code></a>
⬆ Bump pypa/gh-action-pypi-publish from 1.12.4 to 1.13.0 (<a
href="https://redirect.github.com/fastapi/typer/issues/1292">#1292</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/9c5c5437dde460c2cf712df4198ea9eeaab64c28"><code>9c5c543</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/4c2129325a1014241a663f99f482a12af6868afc"><code>4c21293</code></a>
📝 Update release notes</li>
<li><a
href="https://github.com/fastapi/typer/commit/bdf471038d6198bb67638b17e8d03fe0a07045d8"><code>bdf4710</code></a>
⬆ Bump ruff from 0.12.10 to 0.12.11 (<a
href="https://redirect.github.com/fastapi/typer/issues/1283">#1283</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/e8e472d1048c097cc69f5152d8146c4ba4d67a00"><code>e8e472d</code></a>
⬆ [pre-commit.ci] pre-commit autoupdate (<a
href="https://redirect.github.com/fastapi/typer/issues/1288">#1288</a>)</li>
<li><a
href="https://github.com/fastapi/typer/commit/075eb3020706c136c56a1899d8d02d50d022ec9a"><code>075eb30</code></a>
📝 Update release notes</li>
<li>Additional commits viewable in <a
href="https://github.com/fastapi/typer/compare/0.12.3...0.17.4">compare
view</a></li>
</ul>
</details>
<br />


Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

CatEek committed:
review comms + tests fix

pre-commit-ci[bot] committed:
style(pre-commit.ci): auto fixes [...]

CatEek committed:
rm params + todos

CatEek committed:
Merge remote-tracking branch 'origin/iz/feat/microsplit_lightning_api' into iz/feat/microsplit_lightning_api
